### PR TITLE
Updating the description of useCache in queryContext

### PR DESCRIPTION
**Pull request by asrayousuf**
The description of "useCache" in the page: http://druid.io/docs/latest/querying/query-context.html describes it as

>  _Flag indicating whether to leverage the query cache for this query. This may be overridden in the broker or historical node configuration_

However, in actual behavior, the broker or historical node configuration does not always override the configuration that is passed in the query context. This is observed, for instance, when the 
_druid.broker.cache.useCache=true_ and the _useCache is set to false_ in the query context. As per the documentation, the broker configuration should override and caching should be used. However, the actual behavior shows that the query-context setting dominates and the cache usage is disabled.

Hence, the description stating that the useCache can be overridden by the broker/ historical configs does not apply in this case.

### CODE REVIEW

**Comment by gianm** [comment]
Hmm, good point. How about this wording? Just trying to make it as clear as possible…

"When set to `false`, disables reading from the query cache for this query. When set to `true`, Druid uses `druid.broker.cache.useCache` or `druid.broker.cache.populateCache` to determine whether or not to read from the query cache."

And, something similar for populateCache would be great too. It has the same sort of behavior.

**Comment by asrayousuf** [comment]
@gianm : The wordings you proposed are much more coherent. I have incorporated the suggestions. Please do have a look!

**Comment by gianm** [comment]
Thanks @asrayousuf, that looks good to me!
